pandas/io/xml.py

```diff
@@ -1119,6 +1127,22 @@ def read_xml(
     """
     check_dtype_backend(dtype_backend)
 
+    if not any(
```

Comment:
We would want to check if the argument is a string _and_ not any of these things.

For example, `pd.read_xml(123)` should not give a deprecation warning